### DISCUSSION TITLE
WARN:smartdns::server::udp:36: error receiving message on udp_socket:

### DISCUSSION THREAD

**Discussion by c-michail** [mokeyish/smartdns-rs#368] [General]
Hello,
  I am getting a lot of these warnings:
  WARN:smartdns::server::udp:36: error receiving message on udp_socket: An existing connection was forcibly closed by the remote host. (os error 10054)

Any ideas ?

Thank you

**Comment by mokeyish** [comment]
Does this happen when booting? I seem to have encountered this too. When booting, the network is not connected yet. At this time, when starting, resolving the upstream IP will report an error.

**Reply by c-michail** [reply]
No it's continuously

![εικόνα](https://github.com/user-attachments/assets/3222b75b-432f-4142-947b-b53f3c5ebb36)

**Comment by mokeyish** [comment]
Are you running on windows?

https://learn.microsoft.com/en-us/troubleshoot/sql/database-engine/connect/tls-exist-connection-closed

**Reply by c-michail** [reply]
Yes windows 11. I've changed the config to use non TLS (All upstreams are non TLS/SSL UDP) and the same errors are still shown. So it isn't any HTTP/SSL/TLS problem.

Debug log doesn't seem to help either

**Reply by c-michail** [reply]
these articles might be helpful:
https://stackoverflow.com/questions/34242622/windows-udp-sockets-recvfrom-fails-with-error-10054
https://www.cnblogs.com/cnpirate/p/4059137.html
https://stackoverflow.com/questions/35061596/multiclient-udp-server-handling-error-code-10054-properly
https://stackoverflow.com/questions/73787999/rust-how-to-handle-error-sending-message-to-disconnected-client (Rust)

**Reply by c-michail** [reply]
Here's the solution for windows only in rust: 

https://github.com/shadowsocks/shadowsocks-rust/issues/262#issuecomment-633112782
You'll need to import the winapi package too: https://crates.io/crates/winapi

from notes:
Ignoring UdpSocket's WSAECONNRESET error
This is because `UdpSocket::recv_from` may return WSAECONNRESET
if you called `UdpSocket::send_to` a destination that is not existed (may be closed).
It is not an error. Could be ignored completely.
We have to ignore it here because it will crash the server.

I hope you add it soon 😃

**Reply by c-michail** [reply]
i made an issue https://github.com/mokeyish/smartdns-rs/issues/391
